src/D2L.CodeStyle.Analyzers/ApiUsage/System.Collections.Immutable/ImmutableCollectionsAnalyzer.cs

```diff
@@ -32,16 +30,20 @@ CompilationStartAnalysisContext context
 			}
 
 			context.RegisterSyntaxNodeAction(
-				ctx => AnalyzeNewImmutableArray( ctx, immutableArrayType ),
+				ctx => AnalyzeNewImmutableArray(
+					ctx,
+					( ObjectCreationExpressionSyntax )ctx.Node,

```

Comment:
It's more lines, but I like it proximity of `( ObjectCreationExpressionSyntax )ctx.Node` and `ObjectCreationExpressionSyntax node` being in the same method.